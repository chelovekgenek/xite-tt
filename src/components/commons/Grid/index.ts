import { Row, Col } from "antd"

export const Grid = {
  Row,
  Col,
}
