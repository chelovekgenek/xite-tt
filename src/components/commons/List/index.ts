import { List as CList } from "./List"
import { Item as CItem } from "./Item"

export const List = Object.assign(CList, {
  Item: CItem,
})
