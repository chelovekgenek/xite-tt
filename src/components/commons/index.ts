export * from "./Modal"
export * from "./Icon"
export * from "./Grid"
export * from "./List"
