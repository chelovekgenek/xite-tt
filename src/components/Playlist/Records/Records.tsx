import React from "react"
import { connect } from "react-redux"

import { List, Icon } from "components/commons"
import { TAppState } from "store/entities"
import { getList, VideoSelectedActions, getSelected } from "store/entities/video"

interface IStateProps {
  videos: ReturnType<typeof getList>
  selected: ReturnType<typeof getSelected>
}
interface IDispatchProps {
  select: typeof VideoSelectedActions.set
}
interface IProps extends IStateProps, IDispatchProps {}

export const Records: React.FC<IProps> = ({ videos, selected, select }) => (
  <React.Fragment>
    {videos.map(({ id, title }) => (
      <List.Item key={id} onClick={() => select(id)}>
        {selected === id && <Icon type="caret-right" />}
        {title}
      </List.Item>
    ))}
  </React.Fragment>
)

export default connect<IStateProps, IDispatchProps, {}, TAppState>(
  state => ({
    videos: getList(state),
    selected: getSelected(state),
  }),
  {
    select: VideoSelectedActions.set,
  },
)(Records)
