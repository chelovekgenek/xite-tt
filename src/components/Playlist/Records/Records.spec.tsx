import React from "react"
import { shallow } from "enzyme"

import { List } from "components/commons"
import videoMock from "mocks/videos.json"
import { Records } from "./Records"

describe("Records", () => {
  it("renders correctly", () => {
    const comp = shallow(<Records videos={videoMock} selected={""} select={jest.fn() as any} />)
    expect(comp).toMatchSnapshot()
  })
  it("adds icon if selected id is equal to video id", () => {
    const comp = shallow(<Records videos={videoMock} selected={videoMock[0].id} select={jest.fn() as any} />)
    expect(comp).toMatchSnapshot()
  })
  it("calls function with item payload function when list item is clicked", () => {
    const mockFn = jest.fn() as any
    const comp = shallow(<Records videos={videoMock} selected={videoMock[0].id} select={mockFn} />)
    comp
      .find(List.Item)
      .at(0)
      .simulate("click")
    expect(mockFn).toBeCalledWith(videoMock[0].id)
  })
})
