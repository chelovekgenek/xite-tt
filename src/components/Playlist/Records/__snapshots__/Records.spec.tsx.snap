// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Records adds icon if selected id is equal to video id 1`] = `
<Fragment>
  <styled.li
    key="1"
    onClick={[Function]}
  >
    <Icon
      type="caret-right"
    />
    asd
  </styled.li>
</Fragment>
`;

exports[`Records renders correctly 1`] = `
<Fragment>
  <styled.li
    key="1"
    onClick={[Function]}
  >
    asd
  </styled.li>
</Fragment>
`;
